src/resources/listings.js

```diff
@@ -12,13 +12,35 @@ class Listings extends ResourceBase{
     return await this.contractService.getAllListingIds()
   }
 
+  async get(address) {
+    const contractData = await this.contractFn(address, "data")
+    let ipfsHash = this.contractService.getIpfsHashFromBytes32(contractData[1])
+    const ipfsData = await this.ipfsService.getFile(ipfsHash)
+
+    let listing = {
+      address: address,
+      ipfsHash: ipfsHash,
+      sellerAddress: contractData[0],
+      priceWei: contractData[2].toString(),
+      price: this.contractService.web3.fromWei(contractData[2], "ether").toNumber(),
+      unitsAvailable: contractData[3].toNumber(),
+      created: contractData[4].toNumber(),
+      expiration: contractData[5].toNumber(),
+
+      name: ipfsData.data.name,
+      category: ipfsData.data.category,
+      description: ipfsData.data.description,
+      location: ipfsData.data.location,
+      pictures: ipfsData.data.pictures
+    }
+
+    return listing
+  }
+
+  // This method is DEPRCIATED
```

Comment:
Should be "deprecated", and it might help to have an explanation for when to remove the method. 😁